src/sdk/model/vac/ba-company.ts

```diff
@@ -0,0 +1,512 @@
+   * @returns {number}
+   */
+  get managedCloudWorkstations(): number {
+    return this._json.managed_cloud_workstations;
+  }
+
+  /**
+   * Get managed physical servers.
+   * @returns {number}
+   */
+  get managedPhysicalServers(): number {
+    return this._json.managed_physical_servers;
+  }
+
+  /**
+   * Get managed cloud servers.
+   * @returns {number}
+   */
+  get managedCloudServers(): number {
+    return this._json.managed_cloud_servers;
+  }
+
+  /**
+   * Get expiration enabled.
+   * @returns {boolean}
+   */
+  get expirationEnabled(): boolean {
+    return this._json.expiration_enabled;
+  }
+
+  /**
+   * Get expiration date.
+   * @returns {number}
+   */
+  get expirationDate(): number {
+    return this._json.expiration_date;
+  }
+
+  /**
+   * Get total storage quota.
+   * @returns {number}
+   */
+  get totalStorageQuota(): number {
+    return this._json.total_storage_quota;
+  }
+
+  /**
+   * Get used storage quota.
+   * @returns {number}
+   */
+  get usedStorageQuota(): number {
+    return this._json.used_storage_quota;
+  }
+
+  /**
+   * Get endpoint.
+   * @returns {string}
+   */
+  get endpoint(): string {
+    return this._json.endpoint;
+  }
+
+  /**
+   * Get the json representation of this class.
+   * @returns {BaCompanyJson}
+   */
+  get json(): BaCompanyJson {
+    return Object.assign({}, this._json);
+  }
+
+  /**
+   * Get the string representation of this class.
+   * @returns {string}
+   */
+  toString(): string {
+    return JSON.stringify(this._json, undefined, 2);
+  }
+
+  /**
+   * Refreshes the Ba Company data by retrieving it from the API again.
+   * @returns {Promise<BaCompany>} promise that resolves with this object
+   */
+  async refresh(): Promise<BaCompany> {
+    return Iland.getHttp().get(`/vac-companies/${ this.uuid }`).then((response) => {
+      this._json = response.data as BaCompanyJson;
+      return this;
+    });
+  }
+
+  /**
+   * Gets the storage usage for the Ba Company
+   * @param {number} start
+   * @param {number} end
+   * @param {number} limit
+   * @param {PerfIntervalType} interval
+   */
+  async getStorageUsage(start: number, end: number, interval: PerfIntervalType,
+                        limit?: number): Promise<Array<VacPerfSample>> {
+    const requestParams: AxiosRequestConfig = {
+      params: {
+        start: start,
+        end: end,
+        interval: interval
+      }
+    };
+    if (limit) {
+      requestParams.params['limit'] = limit;
+    }
+    return Iland.getHttp().get(`/vac-companies/${ this.uuid }/storage-usage`, requestParams).then((response) => {
+      const samples = response.data.data as Array<VacPerfSampleJson>;
+      return samples.map(it => new VacPerfSample(it));
+    });
+  }
+
+  /**
+   * Gets the backup history for the Ba Company
+   * @param {number} offset
+   * @param {number} limit
+   */
+  async getBackupHistory(offset: number, limit: number): Promise<Array<BaCompanyBackupHistory>> {
```

Comment:
offset and limit should be optional, right?

Reply:
indeed, you're right. Will make them optional then.